core/metrics-spi/src/main/java/software/amazon/awssdk/metrics/SdkMetric.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.metrics;
+
+import java.util.Set;
+import software.amazon.awssdk.annotations.SdkPreviewApi;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.metrics.internal.DefaultSdkMetric;
+
+/**
+ * A specific SDK metric.
+ *
+ * @param <T> The type for values of this metric.
+ *
+ * <b>NOTE:</b> This is a Preview API and is subject to change so it should not be used in production.
+ */
+@SdkPreviewApi
+@SdkPublicApi
+public interface SdkMetric<T> {
+
+    /**
+     * @return The name of this metric.
+     */
+    String name();
+
+    /**
+     * @return The categories of this metric.
+     */
+    Set<MetricCategory> categories();
+
+    /**
+     * @return The level of this metric.
+     */
+    MetricLevel level();
+
+    /**
+     * @return The class of the value associated with this metric.
+     */
+    Class<T> valueClass();
+
+    /**
+     * Cast the given object to the value class associated with this event.
+     *
+     * @param o The object.
+     * @return The cast object.
+     * @throws ClassCastException If {@code o} is not an instance of type {@code
+     * T}.
+     */
+    T convertValue(Object o);
```

Comment:
Do we use this? Why not just let the caller use `valueClass().cast(o)`?

Reply:
Sure this can be removed, don't think it's used at the moment